File: HackerRank/week 2/plus minus.py
import math
import os
import random
import re
import sys


def plusMinus(arr):
    positive = 0
    negative=0
    zero=0
    total =len(arr)
    for num in arr:
        if num>0:
            positive+=1
        elif num<0:
            negative+=1
        else:
            zero+=1
    print(positive/len(arr))
    print(negative/len(arr))
    print(zero/len(arr))

if __name__ == '__main__':
    n = int(input().strip())

    arr = list(map(int, input().rstrip().split()))

    plusMinus(arr)

File: HackerRank/week 1/plus minus.c
#include <stdio.h>
int main()
{
    int n,i;
    int number; 
    int positive=0;
    int negative=0;
    int zero=0;
    scanf("%d",&n);
    for(i=0;i<n;i++)
    {
        scanf("%d",&number);
        if(number>0)
            positive++;
        else if(number<0)
            negative++;
        else
            zero++;
    }
    printf("%.6f\n",(double)positive/n);
    printf("%.6f\n",(double)negative/n);
    printf("%.6f\n",(double)zero/n);
    return 0;
}